### PR TITLE
Authentication part2

### PR DESCRIPTION
**Pull request by impguard**
Second part! There will be one more after this.

Implements the concept of global state within the app so that we don't continuously relogin if the user is still in a session. However, the access token is still not being used in our API.

### CODE REVIEW

File: app/assets/javascripts/shared/appState.coffee [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    this.setUserId = (id) -> user.id = id
+    this.setUserAvatar = (avatar) -> user.avatar = avatar
+    this.setUserName = (name) -> user.username = name
+
+    # Login state
+    Object.defineProperty(this, "loggedIn", {
+        get: () -> loggedIn
+    })
+
+    this.login = () -> loggedIn = false
```
Comment (leohtkam) [inline app/assets/javascripts/shared/appState.coffee:19]: You mean "loggedIn = true"?

File: features/step_definitions/user_profile_steps.rb [hunk 2]
```diff
@@ -16,6 +16,12 @@
 end
 
 Then /^I should see the text "(.*?)" for "(.*?)"$/ do |text, id|
+    sleep(2)
+    puts id
+    puts text
+    puts page.find("##{id}")
+    puts page.find("##{id}").text
+    puts page.text
```
Comment (leohtkam) [inline features/step_definitions/user_profile_steps.rb]: Should remove these.

**Comment by leohtkam** [comment]
+1

File: app/assets/javascripts/shared/appState.coffee (hunk 1 above)
Comment (impguard) [inline app/assets/javascripts/shared/appState.coffee:19]: yea, it's unused so I'm not gonna change it right now. It's removed in the next patch.

**Comment by whsieh** [comment]
+1

**Pull request merged**